src/coordinatorapi.cpp

```diff
@@ -160,6 +160,13 @@ void CoordinatorAPI::resetOnFork(CoordinatorAPI& coordAPI)
   instance()._nsSock.close();
 }
 
+void CoordinatorAPI::resetOnRestart()
+{
+  _coordinatorSocket = jalib::JSocket(PROTECTED_COORD_FD);
+  JASSERT(_coordinatorSocket.isValid()) (JASSERT_ERRNO)
```

Comment:
I have three main comments about this function:

 - I think the assert is not necessary. `isValid()` checks if the fd is >=0, which is trivially true after the code on the previous line. You can continue to use the assert, if you like, but please fix the associated text. This is not a listen socket. :-)

 - Could you please add a comment explaining the purpose of this function?

 - How about renaming the function to: `resetCoordSocketFd()`? I feel like that's more precise.